pkgs/os-specific/linux/procps-ng/default.nix

```diff
@@ -1,4 +1,6 @@
-{ lib, stdenv, fetchurl, ncurses, pkgconfig }:
+{ lib, stdenv, fetchurl, ncurses, pkgconfig
+, systemd ? null
```

Comment:
This is not the correct way to do this. For instance on macOS systems systemd will not be null - it just won't build because it is linux-only. The closest to a correct solution is something like this:

```
{ lib, stdenv, fetchurl, ncurses, pkgconfig
, withSystemd ? stdenv.isLinux, systemd
```